spark/src/main/scala/org/apache/spark/sql/delta/perf/OptimizeMetadataOnlyDeltaQuery.scala

```diff
@@ -60,37 +127,199 @@ trait OptimizeMetadataOnlyDeltaQuery {
     Some(numRecords)
   }
 
-  /** e.g. sql("SELECT COUNT(*) FROM <deta-table>").collect() */
-  object CountStarDeltaTable {
-    def unapply(plan: Aggregate): Option[Long] = plan match {
-      case Aggregate(
-          Nil,
-          Seq(Alias(AggregateExpression(Count(Seq(Literal(1, _))), Complete, false, None, _), _)),
-          PhysicalOperation(_, Nil, DeltaTable(i: TahoeLogFileIndex))) if i.partitionFilters.isEmpty
-        => extractGlobalCount(i)
-      case _ => None
+  val columnStatsSupportedDataTypes: HashSet[DataType] = HashSet(
+    ByteType,
+    ShortType,
+    IntegerType,
+    LongType,
+    FloatType,
+    DoubleType,
+    DateType)
+
+  case class DeltaColumnStat(
+    min: Any,
+    max: Any)
+
+  def extractGlobalColumnStats(tahoeLogFileIndex: TahoeLogFileIndex):
+  CaseInsensitiveMap[DeltaColumnStat] = {
+
+    // TODO Update this to work with DV (https://github.com/delta-io/delta/issues/1485)
+
+    val deltaScanGenerator = getDeltaScanGenerator(tahoeLogFileIndex)
+    val snapshot = deltaScanGenerator.snapshotToScan
+
+    def extractGlobalColumnStatsDeltaLog(snapshot: Snapshot):
+    Map[String, DeltaColumnStat] = {
+
+      val dataColumns = snapshot.statCollectionPhysicalSchema
+        .filter(col => columnStatsSupportedDataTypes.contains(col.dataType))
```

Comment:
can we exit early here if the dataColumns is empty?

Reply:
We do in the next `if` statement. If dataColumns is empty the lazy variables won't be evaluated